pkg/components/sqlprune/postgres.go

```diff
@@ -0,0 +1,103 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package sqlprune
+
+import (
+	"bytes"
+
+	"golang.org/x/sys/unix"
+
+	"go.opentelemetry.io/obi/pkg/app/request"
+)
+
+const (
+	PostgresHdrSize   = 4
+	PostgresErrMinLen = PostgresHdrSize + 6
+
+	PostgresMessageTypeQuery   = 'Q'
+	PostgresMessageTypeBind    = 'B'
+	PostgresMessageTypeExecute = 'E'
+	PostgresMessageTypeParse   = 'P'
+)
+
+func parsePostgresMessageType(buf []uint8) uint8 {
+	if len(buf) < PostgresHdrSize+1 {
+		return 0
+	}
+	// The first byte of the header is the message type
+	return buf[0]
+}
+
+// Postgres error message contains a set of key-value fields, of which
+// 3 are always present: S, C, M and the others are optional:
+//
+// 'E'                                                          // ErrorResponse message type
+// 'S' 'ERROR' '\0'                                             // Severity
+// 'C' '23505' '\0'                                             // SQLSTATE code (unique violation)
+// 'M' 'duplicate key value violates unique constraint "mytable_pkey"' '\0' // Error message
+// 'D' 'Key (id)=(1) already exists.' '\0'                      // Detail
+// 'H' 'Perhaps you need to add a conditional statement?' '\0'  // Hint
+// 'F' 'nbtinsert.c' '\0'                                       // Source file
+// 'L' '402' '\0'                                               // Source line number
+// 'R' '_bt_check_unique' '\0'                                  // Routine
+// '\0'                                                         // End of message
+func parsePostgresError(buf []uint8) *request.SQLError {
```

Comment:
Do you mind adding tests for this file?

Reply:
I was thinking maybe making some simple tests in postgres_test.go and just running various edge cases, maybe it's all covered, but it's hard to tell with the other test. These days any Gen AI tool will give you a good start if you  point it to this code.

Reply:
Done